packages/fxa-content-server/app/scripts/views/choose_what_to_sync.js

```diff
@@ -93,7 +93,10 @@ const View = FormView.extend({
 
     return this.user.setAccount(account)
       .then(account => {
-        this.notifier.trigger('set-sync-engines', offeredSyncEngines);
+        this.notifier.trigger(
+          'set-sync-engines',
+          offeredSyncEngines.filter(e => declinedSyncEngines.indexOf(e) === -1)
```

Comment:
Quadratic nested loop alert!

Reply:
I think this should be fine, would your front end metrics detect any changes with it?

Reply:
No. We're talking about a couple of milliseconds in the worst case.